prov/gni/src/gnix_cq.c

```diff
@@ -327,6 +345,50 @@ ssize_t _gnix_cq_add_error(struct gnix_fid_cq *cq, void *op_context,
 	return ret;
 }
 
+int _gnix_cq_poll_nic_add(struct gnix_fid_cq *cq, struct gnix_nic *nic)
+{
+	struct gnix_cq_poll_nic *pnic, *tmp;
+
+	dlist_for_each_safe(&cq->poll_nics, pnic, tmp, list) {
+		if (pnic->nic == nic) {
+			pnic->ref_cnt++;
```

Comment:
I need to go back and consider locking everywhere.  I may wrap uses of cq->poll_nics with a CQ lock.